src/collectors/uniswapx_route_collector.rs

```diff
@@ -19,13 +17,43 @@ const SLIPPAGE_TOLERANCE: &str = "0.5";
 const DEADLINE: u64 = 1000;
 
 #[derive(Debug, Clone)]
-pub struct OrderData {
-    pub order: ExclusiveDutchOrder,
+pub struct V2DutchOrderData {
```

Comment:
can you just have `OrderData` which uses the order enum type? no need to have enum of enums mapping the same concept

like 
```
pub struct OrderData {
  pub order: Order; // this is enum of V2DutchOrder | PriorityOrder,
  ... shared fields
}


// later, can match on the inner enum
match orderData.order {
   Order::DutchV2ORder =>,
   Order::PriorityORDer =>

}
```